File: chatbot-backend/src/backend/models/document.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class Document(BaseModel):
    title: str  
    pages: int  
    level: str  
    subject: str  
    file_type: str 
    upload_date: datetime = Field(default_factory=datetime.utcnow)  
    file_url: str  

    created_at: Optional[datetime] = Field(default_factory=datetime.utcnow)
    updated_at: Optional[datetime] = Field(default_factory=datetime.utcnow)

File: backend/model/Document.js
const mongoose = require("mongoose");

const DocumentSchema = new mongoose.Schema(
    {
        title: { type: String, required: true },
        pages: { type: Number, required: true },
        level: { type: mongoose.Schema.Types.ObjectId, ref: "Level", required: true },
        subject: { type: mongoose.Schema.Types.ObjectId, ref: "Subject", required: true },
        fileType: { type: String, required: true },
        uploadDate: { type: Date, default: Date.now },
        fileUrl: { type: String, required: true },
    },
    { timestamps: true }
);

const Document = mongoose.model("Document", DocumentSchema);
module.exports = Document;
